modules/media_event_dispatcher/tests/src/Unit/Media/OEmbedResourceUrlAlterEventTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\Tests\media_event_dispatcher\Unit\Media;
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\media\OEmbed\Provider;
+use Drupal\hook_event_dispatcher\HookEventDispatcherInterface;
+use Drupal\media_event_dispatcher\Event\Media\OEmbedResourceUrlAlterEvent;
+use Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Class OEmbedResourceUrlAlterEventTest.
+ *
+ * @package Drupal\Tests\media_event_dispatcher\Unit\Media
+ *
+ * @group hook_event_dispatcher
+ */
+class OEmbedResourceUrlAlterEventTest extends UnitTestCase {
+
+  /**
+   * The manager.
+   *
+   * @var \Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy
+   */
+  private $manager;
+
+  /**
+   * Sets up the test.
+   */
+  public function setUp() {
+    $builder = new ContainerBuilder();
+    $this->manager = new HookEventDispatcherManagerSpy();
+    $builder->set('hook_event_dispatcher.manager', $this->manager);
+    $builder->compile();
+    \Drupal::setContainer($builder);
+  }
+
+  /**
+   * Get a mock oEmbed provider.
+   *
+   * @return \Drupal\media\OEmbed\Provider
+   *   A mock Vimeo oEmbed provider.
+   */
+  protected function getMockProvider() {
```

Comment:
This can be private, no need to be protected.

Reply:
Will do. I guess I'm always erring on the side of making it easier to extend stuff, but it's arguably not necessary here.

Reply:
I prefer to follow SOLID and in this case the O from open–closed principle. This is open for extension, but should be closed for modification.